lib/import/colorectal/providers/salisbury/salisbury_handler_colorectal.rb

```diff
@@ -41,98 +44,232 @@ class SalisburyHandlerColorectal < Import::Brca::Core::ProviderHandler
-            super
-          end
+                                                STK11)/xi.freeze # Added by
 
           def process_fields(record)
             genocolorectal = Import::Colorectal::Core::Genocolorectal.new(record)
             genocolorectal.add_passthrough_fields(record.mapped_fields,
                                                   record.raw_fields,
                                                   PASS_THROUGH_FIELDS_COLO)
-            # add_colorectal_from_raw_test(genocolorectal, record) # Added by Francesco
-            extract_variant(record.raw_fields['genotype'], genocolorectal)
             Maybe(record.raw_fields['moleculartestingtype']).each do |ttype|
-              genocolorectal.add_molecular_testing_type_strict(TEST_TYPE_MAPPING_COLO[ttype.downcase.strip])
+              genocolorectal.add_molecular_testing_type_strict(
+                TEST_TYPE_MAPPING_COLO[
+                  ttype.downcase.strip
+                ]
+              )
               scope = TEST_SCOPE_MAPPING_COLO[ttype.downcase.strip]
               genocolorectal.add_test_scope(scope) if scope
             end
             genocolorectal.add_specimen_type(record.mapped_fields['specimentype'])
             genocolorectal.add_received_date(record.raw_fields['date of receipt'])
-            extract_teststatus(genocolorectal, record)
             add_organisationcode_testresult(genocolorectal)
             res = add_colorectal_from_raw_test(genocolorectal, record)
             res.map { |cur_genotype| @persister.integrate_and_store(cur_genotype) }
-            status = genocolorectal.attribute_map['teststatus']
-            @logger.debug "Extracted Teststatus was #{status}"
           end
 
           def add_organisationcode_testresult(genocolorectal)
             genocolorectal.attribute_map['organisationcode_testresult'] = '699H0'
           end
-          
-          def add_colorectal_from_raw_test(genocolorectal, record)
-            colo_string = record.raw_fields['test']
-            genotypes = []
-            if colo_string.scan(COLORECTAL_GENES_REGEX).size > 1
-              @logger.error "Multiple genes detected in input string: #{colo_string};"
-              colo_string.scan(COLORECTAL_GENES_REGEX).each do |gene|
-                genocolorectal1 = genocolorectal.dup_colo
-                genocolorectal1.add_gene_colorectal(gene[0])
-                genotypes << genocolorectal1
-              end
-            elsif COLORECTAL_GENES_REGEX.match(colo_string) && colo_string.scan(COLORECTAL_GENES_REGEX).size == 1
-              genocolorectal.add_gene_colorectal($LAST_MATCH_INFO[:colorectal])
-              genotypes << genocolorectal
-              @logger.debug "SUCCESSFUL gene change parse for: #{$LAST_MATCH_INFO[:colorectal]}"
-            else
-              @logger.debug "FAILED cdna channge parse for  #{colo_string}"
+
+          def normal_test?(record)
+            return if record.raw_fields['status'].nil?
```

Comment:
@FrancescoNea predicate methods should ideally return true or false, there are a few more examples